gnd/src/main/java/com/google/android/gnd/ui/home/mapcontainer/MapContainerFragment.java

```diff
@@ -131,6 +135,32 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
     mapContainerViewModel
         .getShowMapTypeSelectorRequests()
         .observe(getViewLifecycleOwner(), __ -> showMapTypeSelectorDialog());
+    mapContainerViewModel.getViewMode().observe(getViewLifecycleOwner(), this::updateUI);
+  }
+
+  private void updateUI(Mode mode) {
+    switch (mode) {
+      case DEFAULT:
+        binding.mapTypeBtn.setVisibility(View.VISIBLE);
```

Comment:
Rather than setting the visibility or other dynamic states of view explicitly, we've started moving towards using data bindings on ViewModel. Would it be possible to expose the visibility of these elements in the corresponding ViewModel and binding to them in the layout XML?